### DISCUSSION TITLE
How to mitigate Vulnerability in Base Image of Keycloak 21.0.1 ?

### DISCUSSION THREAD

**Discussion by pdudani09** [keycloak/keycloak#19315] [Archive]
opt/keycloak/lib/lib/main/com.h2database.h2-2.1.214.jar
=======================================================
**Total: 1 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 1, CRITICAL: 0)**

+-------------------+------------------+----------+-------------------+---------------+---------------------------------------+
|      LIBRARY      | VULNERABILITY ID | SEVERITY | INSTALLED VERSION | FIXED VERSION |                 TITLE                 |
+-------------------+------------------+----------+-------------------+---------------+---------------------------------------+
| com.h2database:h2 | CVE-2022-45868   | HIGH     | 2.1.214           |               | Password exposure in H2 Database      |
|                   |                  |          |                   |               | -->avd.aquasec.com/nvd/cve-2022-45868 |
+-------------------+------------------+----------+-------------------+---------------+---------------------------------------+

opt/keycloak/lib/lib/main/org.wildfly.security.wildfly-elytron-1.20.1.Final.jar
===============================================================================
**Total: 4 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 3, CRITICAL: 1)**

+---------------------------------------------+------------------+----------+-------------------+-----------------------------+---------------------------------------+
|                   LIBRARY                   | VULNERABILITY ID | SEVERITY | INSTALLED VERSION |        FIXED VERSION        |                 TITLE                 |
+---------------------------------------------+------------------+----------+-------------------+-----------------------------+---------------------------------------+
| com.fasterxml.jackson.core:jackson-databind | CVE-2022-42003   | HIGH     | 2.13.3            | 2.13.4.1, 2.12.7.1          | jackson-databind: deep                |
|                                             |                  |          |                   |                             | wrapper array nesting wrt             |
|                                             |                  |          |                   |                             | UNWRAP_SINGLE_VALUE_ARRAYS            |
|                                             |                  |          |                   |                             | -->avd.aquasec.com/nvd/cve-2022-42003 |
+                                             +------------------+          +                   +-----------------------------+---------------------------------------+
|                                             | CVE-2022-42004   |          |                   | 2.13.4, 2.12.7.1            | jackson-databind: use                 |
|                                             |                  |          |                   |                             | of deeply nested arrays               |
|                                             |                  |          |                   |                             | -->avd.aquasec.com/nvd/cve-2022-42004 |
+---------------------------------------------+------------------+----------+-------------------+-----------------------------+---------------------------------------+
| org.apache.sshd:sshd-common                 | CVE-2022-45047   | CRITICAL | 2.7.0             | 2.9.2                       | mina-sshd: Java unsafe                |
|                                             |                  |          |                   |                             | deserialization vulnerability         |
|                                             |                  |          |                   |                             | -->avd.aquasec.com/nvd/cve-2022-45047 |
+---------------------------------------------+------------------+----------+-------------------+-----------------------------+---------------------------------------+
| org.wildfly.security:wildfly-elytron        | CVE-2022-3143    | HIGH     | 1.20.1.Final      | 1.20.3.Final, 1.15.15.Final | wildfly-elytron: possible timing      |
|                                             |                  |          |                   |                             | attacks via use of unsafe comparator  |
|                                             |                  |          |                   |                             | -->avd.aquasec.com/nvd/cve-2022-3143  |
+---------------------------------------------+------------------+----------+-------------------+-----------------------------+---------------------------------------+

opt/keycloak/lib/lib/main/org.yaml.snakeyaml-1.33.jar
=====================================================
Total: 1 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 1)

+--------------------+------------------+----------+-------------------+---------------+--------------------------------------+
|      LIBRARY       | VULNERABILITY ID | SEVERITY | INSTALLED VERSION | FIXED VERSION |                TITLE                 |
+--------------------+------------------+----------+-------------------+---------------+--------------------------------------+
| org.yaml:snakeyaml | CVE-2022-1471    | CRITICAL |              1.33 |               | SnakeYaml: Constructor               |
|                    |                  |          |                   |               | Deserialization                      |
|                    |                  |          |                   |               | Remote Code Execution                |
|                    |                  |          |                   |               | -->avd.aquasec.com/nvd/cve-2022-1471 |
+--------------------+------------------+----------+-------------------+---------------+--------------------------------------+

**Comment by abstractj** [comment]
@pdudani09 closing as a duplicate of https://github.com/keycloak/keycloak/discussions/20175
